index.d.ts

```diff
@@ -5,10 +5,41 @@ export class Vec3 {
   y: number;
   z: number;
 
+  /**
+   * Returns true when it is a zero vector.
+   */
+  isZero(): boolean;
+
+  /**
+   * Access component by index
+   */
+  at(id: number): number;
+
+  /**
+   * Returns an array component x, z
+   */
+  xz(): [number, number];
+
+  /**
+   * Returns an array component x, y
+   */
+  xy(): [number, number];
+
+  /**
+   * Returns an array component y, z
+   */
+  yz(): [number, number];
+
+  /**
+   * Returns a vector with swapped y and z
+   */
+  xzy(): Vec3;
+
   /**
    * Set own values to given x y z
+   * If some components is given null, then those components won't change
    */
-  set(x: number, y: number, z: number): this;
+  set(x: number | null, y: number | null, z: number | null): this;
```

Comment:
That's pretty confusing. I think it's better to not have that and instead keep advising the user to set x y or z directly if they don't want to set all three coordinates

Reply:
I guess you are right, maybe it is confusing indeed. removed in d704a6984f6a1dd45cec433bf48d36d55071f229